src/include/primer/hyperloglog_presto.h

```diff
@@ -54,7 +54,17 @@ class HyperLogLogPresto {
    *
    * @returns hash value
    */
-  inline auto CalculateHash(T val) -> hash_t { return std::hash<T>{}(val); }
+  inline auto CalculateHash(T val) -> hash_t {
+    Value val_obj;
+    if constexpr (std::is_same<T, std::string>::value) {
+      val_obj = Value(VARCHAR, val);
+      return bustub::HashUtil::HashValue(&val_obj);
+    }
+    if constexpr (std::is_same<T, int64_t>::value) {
+      return static_cast<hash_t>(val);
```

Comment:
Why don't make it the same as`CalculateHash(T val) -> hash_t` in `hyperloglog.h`?
```

inline auto CalculateHash(T val) -> hash_t {
    Value val_obj;
    if constexpr (std::is_same<T, std::string>::value) {
      val_obj = Value(VARCHAR, val);
    } else {
      val_obj = Value(BIGINT, val);
    }
    return bustub::HashUtil::HashValue(&val_obj);
  }

```

and also there is an issue with MacOS tests in the current master branch

Reply:
For testing special scenarios! Hence, doing the same.

Reply:
The error is fixed now...can clone the latest changes